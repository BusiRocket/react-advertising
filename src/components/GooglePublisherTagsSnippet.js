import PropTypes from 'prop-types';
import React from 'react';

function GooglePublisherTagsSnippet({ active }) {
    const loadGptScriptTag = active ? <script async src="//www.googletagservices.com/tag/js/gpt.js" /> : null;
    return (
        <div>
            {loadGptScriptTag}
            <script
                dangerouslySetInnerHTML={{
                    __html: 'var googletag=googletag||{};googletag.cmd=googletag.cmd||[]'
                }}
            />
        </div>
    );
}

GooglePublisherTagsSnippet.propTypes = {
    active: PropTypes.bool
};

GooglePublisherTagsSnippet.defaultProps = {
    active: true
};

export default GooglePublisherTagsSnippet;
