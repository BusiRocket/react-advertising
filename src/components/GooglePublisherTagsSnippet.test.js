import React from 'react';
import GooglePublisherTagsSnippet from './GooglePublisherTagsSnippet';
import renderer from 'react-test-renderer';

describe('The GooglePublisherTagsSnippet component', () => {
    describe('with no props', () => snapshotTest(<GooglePublisherTagsSnippet />));
    describe('with active prop set', () => snapshotTest(<GooglePublisherTagsSnippet active />));
});

function snapshotTest(component) {
    it('is rendered correctly', () => expect(renderer.create(component).toJSON()).toMatchSnapshot());
}
