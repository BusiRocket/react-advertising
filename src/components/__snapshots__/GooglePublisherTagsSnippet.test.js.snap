// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The GooglePublisherTagsSnippet component with active prop set is rendered correctly 1`] = `
<div>
  <script
    async={true}
    src="//www.googletagservices.com/tag/js/gpt.js"
  />
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "var googletag=googletag||{};googletag.cmd=googletag.cmd||[]",
      }
    }
  />
</div>
`;

exports[`The GooglePublisherTagsSnippet component with no props is rendered correctly 1`] = `
<div>
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "var googletag=googletag||{};googletag.cmd=googletag.cmd||[]",
      }
    }
  />
</div>
`;
